From: =?utf-8?q?Norman_Me=C3=9Ftorff?= <normes@normes.org>
Date: Sun, 23 Mar 2014 17:49:36 +0100
Subject: fix-sql-injection_show_alias

Fix a possible SQL injection of $show_alias in gen_show_status() function.

Upstream commit r1639:
alias can contain a ' - as in email.o'connor - need to therefore escape this...
http://sourceforge.net/p/postfixadmin/code/1639/
---
 functions.inc.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/functions.inc.php b/functions.inc.php
index 7547f0d..2964f86 100644
--- a/functions.inc.php
+++ b/functions.inc.php
@@ -2175,6 +2175,8 @@ function gen_show_status ($show_alias)
     global $CONF, $table_alias;
     $stat_string = "";
 
+    $show_alias = escape_string($show_alias);
+
     $stat_goto = "";
     $stat_result = db_query ("SELECT goto FROM $table_alias WHERE address='$show_alias'");
     if ($stat_result['rows'] > 0)
